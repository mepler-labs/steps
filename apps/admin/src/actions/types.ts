export type DispatchFn = (dispatch?: any, getState?: any) => any;
