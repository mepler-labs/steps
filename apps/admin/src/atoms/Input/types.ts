export type InputType = 'text' | 'textarea';
