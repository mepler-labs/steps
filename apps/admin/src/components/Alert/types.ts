export enum AlertLevel {
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export interface Alert {
  level: AlertLevel;
  message: string;
  id: string;
}
