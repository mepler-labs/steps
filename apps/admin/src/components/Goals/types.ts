export type Goal = {
  text: string;
};
