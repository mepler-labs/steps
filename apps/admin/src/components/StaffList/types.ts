export interface PendingInvite {
  pendingInvite: true;
  email: string;
}

export type Staff<T> = {
  pendingInvite: false;
} & T;

export interface User {
  name: string;
  permissionLevel: PermissionLevel;
  email: string;
}

export enum PermissionLevel {
  Administrator = 'administrator',
  Coach = 'coach',
}

export type StaffMember = PendingInvite | Staff<User>;
