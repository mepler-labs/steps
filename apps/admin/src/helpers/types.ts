export interface HTMLProps {
  className?: string;
}
