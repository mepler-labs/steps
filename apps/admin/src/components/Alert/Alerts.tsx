import { Alert as AlertType } from './types';
import * as React from 'react';
import { Alert } from './Alert';
import styled from 'styled-components';

interface Props {
  alerts: AlertType[];
}

export const Alerts = ({ alerts }: Props) => {
  return (
    <StyledAlerts>
      {alerts.map(({ level, message }, i) => {
        return (
          <Alert key={i} level={level}>
            {message}
          </Alert>
        );
      })}
    </StyledAlerts>
  );
};

const StyledAlerts = styled.div``;

export default Alerts;
