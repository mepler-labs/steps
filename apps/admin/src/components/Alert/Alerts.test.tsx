import { shallow } from 'enzyme';
import 'jest';
import * as React from 'react';
import { Alerts } from './Alerts';
import { AlertLevel } from './types';

describe('Alerts.tsx', () => {
  it('renders correctly', () => {
    const alerts = [
      {
        message: 'some warning',
        level: AlertLevel.Warning
      },
      {
        message: 'some error',
        level: AlertLevel.Error
      }
    ];
    const wrapper = shallow(<Alerts alerts={alerts} />);

    expect(wrapper).toMatchSnapshot();
  });
});
