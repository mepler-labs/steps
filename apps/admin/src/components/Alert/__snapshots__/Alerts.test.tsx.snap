// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alerts.tsx renders correctly 1`] = `
<styled.div>
  <Component
    key="0"
    level="warning"
  >
    some warning
  </Component>
  <Component
    key="1"
    level="error"
  >
    some error
  </Component>
  ;
</styled.div>
`;
